### ISSUE TITLE
`hidden_domain` option fails with HTTPS CONNECT requests, works only with HTTP GET

### ISSUE THREAD

**Issue by dmitry-j-mikhin** [SenseUnit/dumbproxy#82]
**Title:** `hidden_domain` option fails with HTTPS CONNECT requests, works only with HTTP GET

**Description:**  
When running `dumbproxy` with the `hidden_domain=vrn.ru` option, the expected behavior occurs for HTTP GET requests: a 407 Proxy Authentication Required response is returned. However, for HTTPS CONNECT requests to the same domain, a 400 Bad Request response is received instead.  

#### Steps to Reproduce:  
1. Start `dumbproxy` with `hidden_domain=vrn.ru`.  
2. Perform an HTTP GET request via the proxy:  
   ```sh
   curl -i http://vrn.ru --proxy https://example.dumbproxy.io
   ```
   **Expected Result:**  
   ```
   HTTP/1.1 407 Proxy Authentication Required
   Content-Length: 31
   Proxy-Authenticate: Basic realm="dumbproxy"
   Date: Sat, 23 Nov 2024 11:47:59 GMT
   Content-Type: text/plain; charset=utf-8
   
   Proxy authentication required.
   ```

3. Perform an HTTPS CONNECT request via the proxy:  
   ```sh
   curl -i https://vrn.ru --proxy https://example.dumbproxy.io
   ```
   **Actual Result:**  
   ```
   HTTP/1.1 400 Bad Request
   Content-Type: text/plain; charset=utf-8
   X-Content-Type-Options: nosniff
   Date: Sat, 23 Nov 2024 11:52:26 GMT
   Content-Length: 13
   
   curl: (56) CONNECT tunnel failed, response 400
   ```

#### Expected Behavior:  
The `hidden_domain` option should support HTTPS CONNECT requests, returning a 407 Proxy Authentication Required response, similar to HTTP GET requests.  

#### Context:  
Modern browsers default to HTTPS for security, making support for HTTPS CONNECT crucial for `hidden_domain` functionality.  

#### Environment:  
- Proxy: `dumbproxy v1.13.1`
- Client: `curl v8.9.1`
- Proxy configuration: `hidden_domain=vrn.ru`

**Comment by Snawoot** [comment]
Right, it seems HTTPS case gives us `vrn.ru:443` as a host, so it doesn't match. Going to fix it soon!

**Comment by Snawoot** [comment]
Fix released in [v1.13.2](https://github.com/SenseUnit/dumbproxy/releases/tag/v1.13.2).

**Comment by dmitry-j-mikhin** [comment]
Thanks!
